src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/SignatureHelpHandler.cs

```diff
@@ -78,12 +91,15 @@ public async Task<SignatureHelp> HandleRequestAsync(TextDocumentPositionParams r
                 },
             };
 
+            _logger.LogInformation($"Requesting signature help for {projectionResult.Uri}.");
+
             var signatureHelp = await _requestInvoker.ReinvokeRequestOnServerAsync<TextDocumentPositionParams, SignatureHelp>(
                 Methods.TextDocumentSignatureHelpName,
                 projectionResult.LanguageKind.ToContainedLanguageContentType(),
                 textDocumentPositionParams,
                 cancellationToken).ConfigureAwait(false);
 
+            _logger.LogInformation($"Returning {(signatureHelp is null ? "no " : string.Empty)}result.");
```

Comment:
I'd just add an `if` statement; otherwise you're allocating a new string every time here 😉